[Tan Zhan Ming] Duke Increments

TanZhanMing opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

j-lum committed:
Change file mode on `gradle` to be executable

j-lum committed:
Merge pull request #12 from j-lum/gradle+x

Change file mode on `gradle` to be executable (#9)

j-lum committed:
Add configuration for console applications

Gradle defaults to an empty stdin which results in runtime exceptions
when attempting to read from `System.in`. Let's add some sensible
defaults for students who may still need to work with the standard
input stream.

j-lum committed:
Merge pull request #13 from j-lum/javaexec

Add configuration for console applications

TanZhanMing committed:
Level 6 + enums

TanZhanMing committed:
no message

TanZhanMing committed:
Level 7

TanZhanMing committed:
Level 8 (inefficient dateparser class, Work in progress)

TanZhanMing committed:
Merge branch 'branch-Level-7'

TanZhanMing committed:
Merge branch 'branch-Level-8'

TanZhanMing committed:
bug fixes

TanZhanMing committed:
more oop done

TanZhanMing committed:
3 basic JUnit tests done

TanZhanMing committed:
Added javadocs

TanZhanMing committed:
Refactored code to comply with given coding standards.
Stuff changed: Names representing methods must be verbs and written in camelCase.
Try-catch form
Logical units within a block should be separated by one blank line.
The ordering of import statements must be consistent.

Stuff unchanged: Put every class in a package.(package was optional, thus did not do)

TanZhanMing committed:
Rushed level 9 for midnight deadline.
Note to self: change list and find code from input parser to ui class

TanZhanMing committed:
Merge branch 'branch-A-JavaDoc'

TanZhanMing committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/DateParser.java
#	src/main/java/FileWriting.java
#	src/main/java/InputParser.java
#	src/main/java/Ui.java

TanZhanMing committed:
Merge branch 'branch-Level-9'

TanZhanMing commented on src/main/java/Duke.java:
```diff
@@ -11,7 +11,8 @@
     public enum Action {
```
test

TanZhanMing committed:
Merge remote-tracking branch 'origin/gradle'

TanZhanMing committed:
Add Gradle support

TanZhanMing committed:
Update gradle settings

TanZhanMing committed:
Create a base template for GUI. However, something went wrong and main class Duke could not be found.

TanZhanMing committed:
Add basic non-fully-functional GUI to Duke.
GUI does not work for: Error messages
The items in the task list do not carry over in the GUI

ambhinav commented on src/main/java/DateParser.java:
```diff
@@ -0,0 +1,96 @@
+    private static SimpleDateFormat formatter13 = new SimpleDateFormat("dd/MMM/yyyy HH:mm:ss");
+    private static SimpleDateFormat formatter14 = new SimpleDateFormat("dd/MMM/yyyy HH:mm");
+    private static SimpleDateFormat formatter15 = new SimpleDateFormat("dd/MMM/yyyy HHmm");
+    private static SimpleDateFormat formatter16 = new SimpleDateFormat("dd/MMM/yyyy");
```
Nice one, this improves the flexibility of the user input date

ambhinav commented on src/main/java/FileWriting.java:
```diff
@@ -0,0 +1,27 @@
+        File tmp = new File("data/temp.txt");
+        if(!tmp.createNewFile()){
+            Files.delete(Paths.get("data/temp.txt"));
+            tmp.createNewFile();
```
Nice one, makes sense to create a file if user does'nt have alr

ambhinav commented:
Hey Zhan Ming, your code was very easy to follow and you made really good use of the OOP concepts

Good things

- Enum class. I did'nt think of using it but after seeing your code, I realised I should have added to make the commands look cleaner

- Date flexibility. I think it was great to include so many date format options to the user

- Javadocs and coding standard. This made your code very easy to read so thanks lol

To improve:

- I think you could improve the way you handle errors. You could create a special exception for Duke problems. Then when you encounter them during run time the special exception can be thrown with a message. So that means you will only be catching one exception in main, and can trace it back easily to the point where the code breaks

- Packaging. But i guess that is personal cos it helped me organise the classes

Thanks

TanZhanMing committed:
Improve on existing GUI. However some problems remain:
1. GUI DOES NOT EXIT
2. GUI DOES NOT SHOW WELCOME MSG
3. IF LIST HAS MORE THAN 5 ITEMS, UNABLE TO SHOW EVERYTHING

bigjunnn commented on src/main/java/DateParser.java:
```diff
@@ -0,0 +1,96 @@
+
+public class DateParser {
+
+    private static SimpleDateFormat formatter1 = new SimpleDateFormat("dd-MM-yyyy HH:mm:ss");
```
I like how you took flexibility of the date inputs into account.
However, I do think you shouldn't be using this method to enhance flexibility. Unfortunately, I face the same problem and do not have a viable solution. Perhaps you could try to seek help from your tutors or the forum for a more viable solution?

bigjunnn commented on src/main/java/InputParser.java:
```diff
@@ -0,0 +1,102 @@
+        String firstWord;
+
+        firstWord = input.split(" ")[0];
+        switch (firstWord) {
```
I like how you used switch statements here! Makes the code much neater 👍

bigjunnn commented on src/main/java/InputParser.java:
```diff
@@ -0,0 +1,102 @@
+                System.out.println("OOPS!!! Please enter a valid number\n");
+                break;
+            } catch (IndexOutOfBoundsException err) {
+                System.out.println("You only have " + taskList.size() + " tasks, please choose a number from that\n");
```
This seems to be a careless mistake. Remember to indent your lines if it overruns. 😄

bigjunnn commented on src/test/java/InputParserTest.java:
```diff
@@ -20,13 +20,13 @@ public void inputTest() throws IOException {
-        inputParser.actionDeterminer(input5);
-        inputParser.actionDeterminer(input6);
-        inputParser.actionDeterminer(input7);
+        inputParser.determineAction(input1);
```
Nice work! This is a thorough test case and you considered many different cases.

bigjunnn reviewed (commented):
Hi Zhan Ming,

Overall, I think your code is great and you have considered many different areas which I haven't thought of, which is something I could learn from you.

Here's what I thought was good about your code:
- Adherence to coding standards
- Considerations for increasing flexibility of code

Some minor improvements:
- I don't think it's a good idea to increase flexibility of code by defining multiple formatters. Perhaps try to look for alternative solutions?
- Maybe you could consider adding comments to the small logic blocks within your code so others could understand your code more effectively.

All the best for the future enhancements! 😆

TanZhanMing committed:
Add assertions to 'ModifyTaskList' and 'UI' classes

TanZhanMing committed:
Implement InputSplitter Class to aggregate splitting methods from InputParser.
Things left to do:
Aggregate 'list' and 'find' from input parser to a new class 'OutputLister'
Add JavaDoc comments to new classes

TanZhanMing committed:
Refactor existing classes and refine them. Also added some javadoc comments.

TanZhanMing committed:
Update gradle build file with shadowjar

TanZhanMing committed:
Test run

TanZhanMing committed:
Add basic duplicate checking

TanZhanMing committed:
Merge branch 'master' of https://github.com/TanZhanMing/duke

# Conflicts:
#	src/main/java/DuplicateChecker.java
#	src/main/java/TaskListModifier.java

TanZhanMing committed:
Iron out git issues

TanZhanMing committed:
Update detection of duplicates. Unable to enhance due to GUI constraints.

TanZhanMing committed:
Adds a representitive screenshot and changes the build.gradle to open UI by default

TanZhanMing committed:
Set theme jekyll-theme-modernist

TanZhanMing committed:
Do a mini-release to tag with A-JAR

TanZhanMing committed:
Merge branch 'master' of https://github.com/TanZhanMing/duke

TanZhanMing committed:
Add code quality to 'DuplicateChecker' class. (#6)

TanZhanMing committed:
Add assertion to new 'DuplicateChecker' class. (#5)

TanZhanMing committed:
Merge branch 'master' of https://github.com/TanZhanMing/duke

TanZhanMing committed:
Slight edit to user guide

TanZhanMing committed:
Change dialogbox text to be displayed by a text field instead of a label.
Update representative screenshot.

TanZhanMing committed:
Update UG slightly

TanZhanMing committed:
Set theme jekyll-theme-minimal

TanZhanMing committed:
test

TanZhanMing committed:
Merge branch 'master' of https://github.com/TanZhanMing/duke

TanZhanMing committed:
test

TanZhanMing committed:
Add a prototype User Guide

TanZhanMing committed:
Finialise User Guide

TanZhanMing committed:
test

TanZhanMing committed:
Update UG

TanZhanMing committed:
Modify 'build.gradle' for v0.2

TanZhanMing committed:
Fix UG typos

TanZhanMing committed:
Update build.gradle for cross-platform compatibility

TanZhanMing committed:
Update duke for cross-platform again

TanZhanMing committed:
Update for cross-platform
